core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala

```diff
@@ -113,22 +113,29 @@ class InvokerPool(childFactory: (ActorRefFactory, InvokerInstanceId) => ActorRef
 
   // State of the actor. Mutable vars with immutable collections prevents closures or messages
   // from leaking the state for external mutation
-  var instanceToRef = immutable.Map.empty[InvokerInstanceId, ActorRef]
+  var instanceToRef = immutable.Map.empty[Int, ActorRef]
   var refToInstance = immutable.Map.empty[ActorRef, InvokerInstanceId]
   var status = IndexedSeq[InvokerHealth]()
 
   def receive: Receive = {
     case p: PingMessage =>
-      val invoker = instanceToRef.getOrElse(p.instance, registerInvoker(p.instance))
-      instanceToRef = instanceToRef.updated(p.instance, invoker)
+      val invoker = instanceToRef.getOrElse(p.instance.toInt, registerInvoker(p.instance))
+      instanceToRef = instanceToRef.updated(p.instance.toInt, invoker)
+
+      // For the case when the invoker was restarted and got a new displayed name
+      val oldHealth = status(p.instance.toInt)
+      if (oldHealth.id != p.instance) {
+        status = status.updated(p.instance.toInt, new InvokerHealth(p.instance, oldHealth.status))
+        instanceToRef.get(p.instance.toInt).map(ref => refToInstance.updated(ref, p.instance))
```

Comment:
You already have the invoker reference above in `invoker`. Therefore you can simplify this line to:

```scala
refToInstance.updated(invoker, p.instance)
```